src/test/java/bio/terra/janitor/service/cleanup/FlightSchedulerTest.java

```diff
@@ -0,0 +1,78 @@
+package bio.terra.janitor.service.cleanup;
+
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.generated.model.GoogleBucketUid;
+import bio.terra.janitor.app.Main;
+import bio.terra.janitor.app.configuration.PrimaryConfiguration;
+import bio.terra.janitor.db.*;
+import com.google.common.collect.ImmutableMap;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@Tag("unit")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
+public class FlightSchedulerTest {
+
+  @Autowired FlightScheduler flightScheduler;
+  @Autowired PrimaryConfiguration primaryConfiguration;
+  @Autowired JanitorDao janitorDao;
+
+  @Test
+  public void resourceScheduledForCleanup() throws Exception {
+    Instant now = Instant.now();
+    TrackedResource resource =
+        TrackedResource.builder()
+            .trackedResourceId(TrackedResourceId.create(UUID.randomUUID()))
+            .trackedResourceState(TrackedResourceState.READY)
```

Comment:
I thought when we create resource, we don't need to pass this value, this is done inside dao's `createResource` method

Reply:
Not since PR#21. I'm moving towards making the JanitorDao dumber and just do the operations of retrieving/modifying the persistent state. That resources often start in the READY state is business logic. (Note that they may not start in the READY 
state if they duplicate another resource. I don't think we want all of the dedupe logic to live in the JanitorDao). Along those lines, I think I'm going to pull the logic out of JanitorDao#updateResourceForCleaning into the FlightMonitor in a follow up PR.